dowhy/gcm/independence_test/ablation.py

```diff
@@ -0,0 +1,107 @@
+""" Ablation based (conditional) independence test. Testing independence via regression, i.e. if a variable has
+information about another variable, then they are dependent.
+"""
+from typing import Callable, Optional, Union, List
+
+import numpy as np
+from sklearn.kernel_approximation import Nystroem
+from sklearn.preprocessing import scale
+
+from dowhy.gcm.stats import quantile_based_fwer, estimate_f_test_p_value
+from dowhy.gcm.util.general import shape_into_2d, apply_one_hot_encoding, fit_one_hot_encoders
+
+
+def ablation_based(X: np.ndarray,
```

Comment:
I see the motivation behind using `ablation`, but not sure if it is a common-enough term. Compared to other tests, the defining factor here is the use of a prediction model.
How about `prederror_based_independence` or `regression_based_independence`?

If you prefer to go with ablation, then I suggest "ablation_based_independence", since the intent of the function is not clear.

Reply:
Agree, I will change it to ```regression_based``` (to be consistent with the naming pattern of the other tests).